File: KotlinCoroutines/app/src/main/java/com/emedinaa/kotlincoroutines/data/http/LoggingInterceptor.java
package com.emedinaa.kotlincoroutines.data.http;

import android.util.Log;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

class LoggingInterceptor implements Interceptor {
    @Override public Response intercept(Interceptor.Chain chain) throws IOException {
        Request request = chain.request();

        long t1 = System.nanoTime();
        Log.i("HTTP",String.format("Sending request %s on %s%n%s",
                request.url(), chain.connection(), request.headers()));

        Response response = chain.proceed(request);

        long t2 = System.nanoTime();
        Log.i("HTTP",String.format("Received response for %s in %.1fms%n%s",
                response.request().url(), (t2 - t1) / 1e6d, response.body()));

        return response;
    }
}

File: KotlinCoroutines/app/src/main/java/com/emedinaa/kotlincoroutines/data/http/LoggingInterceptor.kt
package com.emedinaa.kotlincoroutines.data.http

import android.util.Log
import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException

/**
 * @author Eduardo Medina
 */
private const val TAG = "HTTP"

class LoggingInterceptor : Interceptor {
    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val request = chain.request()

        val t1 = System.nanoTime()
        Log.i(TAG, "Sending request ${request.url} on ${chain.connection()} ${request.headers}")

        val response = chain.proceed(request)

        val t2 = System.nanoTime()
        Log.i(
            TAG,
            "Received response for ${response.request.url} in ${(t2 - t1) / 1e6} ${response.body}"
        )
        return response
    }
}